superset/assets/src/dashboard/components/gridComponents/ChartHolder.jsx

```diff
@@ -81,32 +84,51 @@ class ChartHolder extends React.Component {
     super(props);
     this.state = {
       isFocused: false,
+      outlinedComponentId: null,
+      outlinedColumnName: null,
+      directPathLastUpdated: 0,
     };
+    // window.clearTimeout's key is global,
+    // we don't want one chart clear out another's chart's timeout.
+    this.timer = [];
 
     this.handleChangeFocus = this.handleChangeFocus.bind(this);
     this.handleDeleteComponent = this.handleDeleteComponent.bind(this);
     this.handleUpdateSliceName = this.handleUpdateSliceName.bind(this);
   }
 
-  getChartAndLabelComponentIdFromPath() {
-    const { directPathToChild = [] } = this.props;
-    const result = {};
+  UNSAFE_componentWillReceiveProps(nextProps) {
```

Comment:
instead of adding a new UNSAFE function, can we use one of the other recommendations here? https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops

Reply:
now use `getDerivedStateFromProps(props, state)`. Thanks!